src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsClientModelValidatorOfTAttribute.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// A generic implementation of <see cref="DataAnnotationsClientModelValidator"/>.
+    /// </summary>
+    /// <typeparam name="TAttribute">An attribute which implements <see cref="ValidationAttribute"/>.</typeparam>
+    public class DataAnnotationsClientModelValidator<TAttribute> : DataAnnotationsClientModelValidator
```

Comment:
what's the scenario where a validator would inherit from `DataAnnotationsClientModelValidator` and not `DataAnnotationsClientModelValidator<TAttribute>`?  we have a few cases of that today but they look like minor bugs.

if this scenario doesn't exist, could remove the base `DataAnnotationsClientModelValidator` class.